features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/DashDirectUserAuthFragment.kt

```diff
@@ -0,0 +1,220 @@
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.launch
+import org.dash.wallet.common.data.ResponseResource
+import org.dash.wallet.common.ui.viewBinding
+import org.dash.wallet.common.util.safeNavigate
+import org.dash.wallet.features.exploredash.R
+import org.dash.wallet.features.exploredash.databinding.FragmentDashDirectUserAuthBinding
+import org.slf4j.LoggerFactory
+
+@FlowPreview
+@ExperimentalCoroutinesApi
+@AndroidEntryPoint
+class DashDirectUserAuthFragment : Fragment(R.layout.fragment_dash_direct_user_auth) {
+    companion object {
+        private val log = LoggerFactory.getLogger(DashDirectUserAuthFragment::class.java)
+    }
+
+    private var currentDirectUserAuthType: DashDirectUserAuthType?=null
+    private val binding by viewBinding(FragmentDashDirectUserAuthBinding::bind)
+    private val exploreViewModel: ExploreViewModel by navGraphViewModels(R.id.explore_dash) { defaultViewModelProviderFactory }
+
+    enum class DashDirectUserAuthType(
+        @StringRes val screenTitle: Int,
+        @StringRes val screenSubtitle: Int,
+        @StringRes val textInputHint: Int
+    ) {
+        CREATE_ACCOUNT(
+            R.string.create_dash_direct_account,
+            R.string.log_in_to_dashdirect_account_desc,
+            R.string.email
+        ),
+        SIGN_IN(
+            R.string.log_in_to_dashdirect_account,
+            R.string.log_in_to_dashdirect_account_desc,
+            R.string.email
+        ),
+        OTP(
+            R.string.enter_one_time_password,
+            R.string.check_your_email_and_enter_the_one_time_password,
+            R.string.password
+        );
+    }
+
+    @RequiresApi(Build.VERSION_CODES.R)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.titleBar.setNavigationOnClickListener {
+            hideKeyboard()
+
+            findNavController().popBackStack()
+        }
+        binding.continueButton.isEnabled = false
+
+        binding.input.postDelayed({
+               showKeyboard()
+            },
+            100
+        )
+
+        currentDirectUserAuthType=   (arguments?.getSerializable("dashDirectUserAuthType") as? DashDirectUserAuthType)?.also {
+             binding.title.setText(it.screenTitle)
+             binding.descLabel.setText(it.screenSubtitle)
+             binding.inputWrapper.setHint(it.textInputHint)
+        }
+
+        binding.input.doOnTextChanged { text, _, _, _ ->
+
+            binding.inputWrapper.isErrorEnabled = false
+
+            if(currentDirectUserAuthType!=DashDirectUserAuthType.OTP )
+            binding.continueButton.isEnabled =isEmail(text)
+            else
+                binding.continueButton.isEnabled = !text.isNullOrEmpty()
```

Comment:
This code snippet is a mess. There are missing and extra spaces and indentations do not match, all of which is making it hard to read. 

Please reformat it manually or use some formatting tool. There are many other places in the code that have similar issues.